packages/cdktf/lib/testing/matchers.ts

```diff
@@ -0,0 +1,179 @@
+import * as fs from "fs";
+import * as path from "path";
+import { execSync } from "child_process";
+
+const terraformBinaryName = process.env.TERRAFORM_BINARY_NAME || "terraform";
+
+export interface TerraformConstructor {
+  new (...args: any[]): any;
+  terraformResourceType: string;
+}
+export type SynthesizedStack = {
+  resource: Record<string, any>;
+  data: Record<string, any>;
+};
+export type MatcherReturn = { message: () => string; pass: boolean };
+
+function assertElementWithProperties(
+  type: keyof SynthesizedStack,
+  received: string,
+  itemType: TerraformConstructor,
+  properties: Record<string, any> = {}
+) {
+  const items = Object.entries(
+    (JSON.parse(received) as SynthesizedStack)[type] || {}
```

Comment:
This could fail more gracefully if something is used with the matcher that is not a string containing valid json